OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseTabsDashboardFragment.java

```diff
@@ -82,12 +84,15 @@ public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
-        if (courseData.getCourse().getCoursewareAccess().hasAccess()) {
-            return super.onCreateView(inflater, container, savedInstanceState);
-        } else {
+        final boolean auditAccessExpired = courseData.getAuditAccessExpires() != null &&
+                new Date().after(DateUtil.convertToDate(courseData.getAuditAccessExpires()));
+
+        if (auditAccessExpired || !courseData.getCourse().getCoursewareAccess().hasAccess()) {
```

Comment:
⚠️ @miankhalid 
This logic should be corrected, as per slack discussion our course content access should only be controlled with `has_access` flag and not with `audit_access_expires` flag.
https://edx-internal.slack.com/archives/CD394ECCU/p1544115918056900?thread_ts=1544114912.051700&cid=CD394ECCU

Also these details should be documented somewhere, story description will be good place too.